lib/api/apiUtils/bucket/models/ReplicationConfiguration.js

```diff
@@ -92,6 +93,25 @@ class ReplicationConfiguration {
     }
 
     /**
+     * Generate a 48-character alphanumeric, unique ID for a rule
+     * @param {number} [n] - The retry count, if calling recursively
+     * @return {string} - The unique rule ID
+     */
+    _createUniqueID(n) {
+        // If no ID is given in the replication configuration, we create it.
+        const id = Buffer.from(UUID.v4()).toString('base64');
+        // Each ID in a list of rules must be unique. If the generated ID
+        // already exists, call recursively until a unique ID is generated or
+        // we have retried five times.
+        let retries = n || 0;
```

Comment:
Isn't UUID supposed to create supposedly unique IDs already?

All in all, I would rather remove the existence checks, it's not worth the added complexity IMO, a simple call to `UUID.v4()` without any further check should be enough.